src/Microsoft.ServiceFabric.Services.Remoting/Description/InterfaceDescription.cs

```diff
@@ -67,10 +67,13 @@ public MethodDescription[] Methods
             MethodReturnCheck methodReturnCheck)
         {
             EnsureValidMethods(remotedInterfaceKindName, remotedInterfaceType, methodReturnCheck);
-            var methods = remotedInterfaceType.GetMethods().Select(
-                methodInfo => MethodDescription.Create(remotedInterfaceKindName, methodInfo)).ToList();
-
-            return methods.ToArray();
+            var methods = remotedInterfaceType.GetMethods();
```

Comment:
The performance test I wrote is very simple:
```
    public interface ITestService : IService
    {
        Task<string> Test1Async(object o, int[] i);
        Task<int> Test2Async(object o, int[] i);
        Task<long> Test3Async(object o, int[] i);
        Task<Guid> Test4Async(object o, int[] i);
        Task<object> Test5Async(object o, int[] i);
    }
```

    [TestClass]
    public class PerformanceBenchmarks
    {
        [TestMethod]
        public void DescriptionCreationBenchmark()
        {
            var stopwatch = new Stopwatch();
            stopwatch.Start();

            for (int i = 0; i < 100000; i++)
            {
                var description = ServiceInterfaceDescription.Create(typeof (ITestService));
            }

            stopwatch.Stop();
        }
    }

In my benchmarks, this led to 8-10% reduction in time for the ServiceInterfaceDescription.Create call. Each individual call is not expensive, but we can utilize memory and CPU slightly more optimally without convoluted code.

![image](https://cloud.githubusercontent.com/assets/10815179/26436072/ab9c034a-40c8-11e7-9988-4458b353db30.png)